librz/core/cmd_analysis.c

```diff
@@ -9721,3 +9157,401 @@ RZ_IPI RzCmdStatus rz_analysis_function_vars_sp_setref_handler(RzCore *core, int
+	return rz_analysis_xrefs_set(core->analysis, from, to, type) ? RZ_CMD_STATUS_OK : RZ_CMD_STATUS_ERROR;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_xrefs_set_0_handler(RzCore *core, int argc, const char **argv) {
+	return xrefs_set(core, argc, argv, RZ_ANALYSIS_REF_TYPE_NULL);
+}
+
+RZ_IPI RzCmdStatus rz_analysis_xrefs_set_c_handler(RzCore *core, int argc, const char **argv) {
+	return xrefs_set(core, argc, argv, RZ_ANALYSIS_REF_TYPE_CODE);
+}
+
+RZ_IPI RzCmdStatus rz_analysis_xrefs_set_C_handler(RzCore *core, int argc, const char **argv) {
+	return xrefs_set(core, argc, argv, RZ_ANALYSIS_REF_TYPE_CALL);
+}
+
+RZ_IPI RzCmdStatus rz_analysis_xrefs_set_d_handler(RzCore *core, int argc, const char **argv) {
+	return xrefs_set(core, argc, argv, RZ_ANALYSIS_REF_TYPE_DATA);
+}
+
+RZ_IPI RzCmdStatus rz_analysis_xrefs_set_s_handler(RzCore *core, int argc, const char **argv) {
+	return xrefs_set(core, argc, argv, RZ_ANALYSIS_REF_TYPE_STRING);
+}
+
+static void xrefs_list_print(RzCore *core, RzList *list) {
+	RzListIter *iter;
+	RzAnalysisXRef *xref;
+
+	rz_list_foreach (list, iter, xref) {
+		char *name = core->analysis->coreb.getNameDelta(core->analysis->coreb.core, xref->from);
+		if (name) {
+			rz_str_replace_ch(name, ' ', 0, true);
+			rz_cons_printf("%40s ", name);
+			free(name);
+		} else {
+			rz_cons_printf("%40s ", "?");
+		}
+		rz_cons_printf("0x%08" PFMT64x " -> %7s -> 0x%08" PFMT64x, xref->from, rz_analysis_xrefs_type_tostring(xref->type), xref->to);
+		name = core->analysis->coreb.getNameDelta(core->analysis->coreb.core, xref->to);
+		if (name) {
+			rz_str_replace_ch(name, ' ', 0, true);
+			rz_cons_printf(" %s\n", name);
+			free(name);
+		} else {
+			rz_cons_printf("\n");
+		}
+	}
+}
+
+static void xrefs_list_print_as_json(RzCore *core, RzList *list, PJ *pj) {
+	RzListIter *iter;
+	RzAnalysisXRef *xref;
+
+	pj_a(pj);
+	rz_list_foreach (list, iter, xref) {
+		pj_o(pj);
+		pj_kn(pj, "from", xref->from);
+		pj_kn(pj, "to", xref->to);
+		pj_ks(pj, "type", rz_analysis_xrefs_type_tostring(xref->type));
+		pj_end(pj);
+	}
+	pj_end(pj);
+}
+
+static const char *xrefs_type2cmd(RzAnalysisXRefType type) {
+	switch (type) {
+	case RZ_ANALYSIS_REF_TYPE_CODE:
+		return "axc";
+	case RZ_ANALYSIS_REF_TYPE_CALL:
+		return "axC";
+	case RZ_ANALYSIS_REF_TYPE_DATA:
+		return "axd";
+	case RZ_ANALYSIS_REF_TYPE_STRING:
+		return "axs";
+	case RZ_ANALYSIS_REF_TYPE_NULL:
+		return "ax";
+	}
+	return "ax";
+}
+
+static void xrefs_list_print_as_cmd(RzCore *core, RzList *list) {
+	RzListIter *iter;
+	RzAnalysisXRef *xref;
+
+	rz_list_foreach (list, iter, xref) {
+		rz_cons_printf("%s 0x%" PFMT64x " @ 0x%" PFMT64x "\n", xrefs_type2cmd(xref->type), xref->to, xref->from);
+	}
+}
+
+RZ_IPI RzCmdStatus rz_analysis_xrefs_list_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	RzListIter *iter;
+	RzAnalysisXRef *xref;
+	if (state->mode == RZ_OUTPUT_MODE_JSON && !state->d.pj) {
+		return RZ_CMD_STATUS_ERROR;
+	}
```

Comment:
you can probably remove this. Or convert it into a `rz_return_val_if_fail`, since it is expected this function is always called with the proper things initialized. In particular, if mode == JSON || LONG_JSON, d.pj should be initialized. Same thing for TABLE.